internal/provisioner/plain/pkg/git/git.go

```diff
@@ -0,0 +1,78 @@
+package git
```

Comment:
One issue that the location of this implementation raises is that it is coupled fairly closely to the plain provisioner's use of an unpack pod, which on the surface makes it less suitable for use in other provisioners.

This reminds me that we may want to create a Source interface that hides the details of the source from the provisioner so that other provisioners could easily reuse the implementation.